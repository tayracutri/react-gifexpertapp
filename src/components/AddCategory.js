import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')
    const handleInput = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {

            setCategories(cat => [inputValue, ...cat]);
            setinputValue('');
        }
    }

    return (

        <form onSubmit={handleSubmit}>

            <input type='text'
                value={inputValue}
                onChange={handleInput}
            />

        </form>

    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
