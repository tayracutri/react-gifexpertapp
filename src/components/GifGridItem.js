import React from 'react';



export const GifGridItem = ({ title, url }) => {


    return (
        <div className='card animate__animated fadeIn'>

            <img src={url} alt={title} />
            <p> {title}</p>


        </div >)

}
