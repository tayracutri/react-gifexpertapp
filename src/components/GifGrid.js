import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types'
import { useFecthGifs } from './hooks/useFecthGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFecthGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeInDown"> {category} </h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="grid-card">

                {data.map(img => (
                    <GifGridItem key={img.id} {...img} />

                ))
                }
            </div>

        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}