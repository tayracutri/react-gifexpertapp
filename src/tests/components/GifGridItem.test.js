import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';




describe('test on gifgriditem', () => {

    const title = 'hi'
    const url = 'https://nojpx'
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('should show the component', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('should have a paragraph w the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('should show an img with src and alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title);

    });



    test('should have an animation', () => {
        const div = wrapper.find('div');
        expect(div.prop('className')).toBe('card animate__animated fadeIn')
    })




});
