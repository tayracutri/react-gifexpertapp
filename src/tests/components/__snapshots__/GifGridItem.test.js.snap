// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on gifgriditem should show the component 1`] = `ShallowWrapper {}`;
